### PR TITLE
fix issue #163

### LINKED ISSUE #163: Refactoring `Root_properties.m`
The file [Root_properties.m](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/src/Root_properties.m) has a few problems:

- Documentation is missing:
  - what does this function calculate?
  - what happens in the if-statements and loops?
- Three global variables are used. These should instead be passed to the function
- A lot of commented out code is in the file. If this code is unused it should be removed.

Additionally, it would be good to also rename the variables/structures according to the [style guide](http://cnl.sogang.ac.kr/cnlab/lectures/programming/matlab/Richard_Johnson-MatlabStyle2_book.pdf) where possible.

### PR DESCRIPTION
**Pull request by DanyangYu**
### Description
Refator the Root_property fuction

### Update executable file
Please re-generate exe file if matlab codes are changed. About how to create an exe file, see [exe readme](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/exe/README.md).

### Checklist
- [x] Add a reference to related issues.
- [x] @mentions of the person or team responsible for reviewing proposed changes.
- [x] This pull request has a descriptive title.
- [x] Code is written according to the [guidelines](http://cnl.sogang.ac.kr/cnlab/lectures/programming/matlab/Richard_Johnson-MatlabStyle2_book.pdf).
- [x] The checks by [MISS_HIT style checker and linter](https://github.com/EcoExtreML/STEMMUS_SCOPE/blob/main/CONTRIBUTING.md#follow-matlab-style-guidelines), below the pull request, are successful (green).
- [x] Documentation is available.
- [x] Model runs successfully.

closes #163

### CODE REVIEW

**Comment by geek-yang** [comment]
It would be nice to have a descriptive title of your PR.

File: src/ebal.m [hunk 1]
```diff
@@ -461,11 +461,11 @@
...
-    %%%%%%%%%%%%%%%%%%% [Delta_Rltot] = Root_properties(Actot,rroot);
-    %%%%%%%%%%%%%%%%%%% Delta_Rl = fc*Delta_Rltot;
-    %%%%%%%%%%%%%%%%%%% Rl = Rl + Delta_Rl;
-    %%%%%%%%%%%%%%%%%%% Rltot = sum(sum(Rl));
-    %%%%%%%%%%%%%%%%%%% fc = Rl./Rltot;
+    % [Delta_Rltot] = Root_properties(Rl, Ac, rroot, frac, bbx, KT, DeltZ, sfactor, LAI_msr);
+    % Delta_Rl = fc*Delta_Rltot;
+    % Rl = Rl + Delta_Rl;
+    % Rltot = sum(sum(Rl));
+    % fc = Rl./Rltot;
```
Comment (geek-yang) [inline src/ebal.m:468]: Why not deleting these lines of code? Are they kept for a reason?

**Review by geek-yang — COMMENTED** [review]
Nice work! I left some comments and address them. Thanks a lot 😄 !

**Review by EntingTang — COMMENTED** [review]
Good to add references, explain the if-statements and loops, make codes clean. 
Possible improvement:  meaning of input and output variables should be explained

File: src/ebal.m (hunk 1 above)
Comment (DanyangYu) [reply src/ebal.m:468]: ok, I will delete them.

File: src/Root_properties.m [hunk 2]
```diff
@@ -3,43 +3,27 @@
...
-    % fr=0.02;
-    % end
+%{
+    This function is used to calculate the dynamic growth of root
+    This part can refer to Wang et al. (2021) "Intergrated modelling of photosynthesis and transfer of energy, mass, and momentum in the SPAC system"
+%}
+
+function [Rl] = Root_properties(Rl, Ac, rroot, frac, bbx, KT, DeltZ, sfactor, LAI_msr)
+    %INPUTS
+    %global DeltZ sfactor LAI_msr
```
[the remark this answers is not in the document]
Comment (DanyangYu) [reply src/Root_properties.m:13]: ok

File: src/Root_properties.m [hunk 3]
```diff
@@ -3,43 +3,27 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 %% REFERENCES
```
Comment (Crystal-szj) [inline src/Root_properties.m:5]:
```suggestion

```

File: src/Root_properties.m (hunk 2 above)
Comment (Crystal-szj) [inline src/Root_properties.m:13]:
```suggestion
function [Rl] = Root_properties(Rl, Ac, rroot, frac, bbx, KT, DeltZ, sfactor, LAI_msr)
%{
    This function is used to calculate the dynamic growth of root
    This part can refer to Wang et al. (2021) "Intergrated modelling of photosynthesis and transfer of energy, mass, and momentum in the SPAC system"
%}
```

Comment (Crystal-szj) [reply src/Root_properties.m:13]: It would be nice if we can add an explanation for the input and output parameters.

File: src/ebal.m (hunk 1 above)
Comment (Crystal-szj) [reply src/ebal.m:468]: If we delete these lines of code, the `Root_properties` will never be called, which means that we delete the root dynamic growth module in our model. Could we add an optional index to decide if we open the root dynamic simulation?

**Review by Crystal-szj — COMMENTED** [review]
Nice to add the reference and explanations for the function and variables 👍. Please see my comments in the code.

**Review by Crystal-szj — APPROVED** [review]
Thanks Danyang, nice work!

**Comment by SarahAlidoost** [comment]
@DanyangYu and @Crystal-szj Thank you for working on the pull request. I noticed that you approved and merged the pull request while the checks were not green. This will reduce the quality of the codes in the main branch. For future pull requests, please make sure that the checks are green before approving and merging them.

**Comment by DanyangYu** [comment]
Got it

**Pull request merged**